Login social first

escapemanuele opened the pull request:
With this PR we will implement 4lWBkrFl1jQuN6irMuoLys-fi-286%3A713.

## Before

<img width="800" alt="image" src="https://github.com/Automattic/wp-calypso/assets/52076348/1868f149-9feb-4773-bb22-0960fcf20e3a">

## After

<img width="800" alt="image" src="https://github.com/Automattic/wp-calypso/assets/52076348/563ebdad-293f-4f65-9dbc-0519698661b1">

## Setting up the Experiment

Follow this (_Testing logged-out homepage implementations_): PCYsg-SwK-p2
With this experiment: 21539-explat-experiment

## Testing

1. Use the live link.
2. Visit `/log-in`. Test desktop and mobile.
3. Visit `/log-in/es` and live link version. Test desktop and mobile.
4. Visit `wordpress.com/start/hosting` and live link version. Test desktop and mobile. They should be identical.
5. Visit `wordpress.com/start/hosting/de` and live link version. Test desktop and mobile. They should be identical.
6. Visit `wordpress.com/start` and live link version. Test desktop and mobile. They should be identical.
7. Visit `wordpress.com/start/de` and live link version. Test desktop and mobile. They should be identical.
8. Visit `wordpress.com/start/videopress-account/user` and live link version. Test desktop and mobile.
9. **P2**: [Link](https://wordpress.com/log-in?redirect_to=http%3A%2F%2Fcalypso.localhost%3A3000%2Fstart%2Fp2%2Fp2-confirm-email&from=p2&signup_url=%2Fstart%2Fp2%2Fuser)  and live link version. Test desktop and mobile.
10. **Jetpack**: [Link](http://wordpress.com/log-in/jetpack?site=https%3A%2F%2Ffalse-shrike.jurassic.ninja)  and live link version. Test desktop and mobile.
11. **Crowdsignal**: [Link](http://wordpress.com/log-in?client_id=978&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D978%26response_type%3Dcode%26blog_id%3D0%26state%3D51d5e1b0d7358dee3de65c45f4b4dc37a3d6e37dab8d6047545f636c2659d104%26redirect_uri%3Dhttps%253A%252F%252Fapp.crowdsignal.com%252Fconnect%253Fsource%253Dlogin%2526action%253Drequest_access_token%26wpcom_connect%3D1) and live link version. Test desktop and mobile.
12. **Jetpack Cloud**: [Link](http://wordpress.com/log-in?client_id=69040&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fresponse_type%3Dtoken%26client_id%3D69040%26redirect_uri%3Dhttps%253A%252F%252Fcloud.jetpack.com%252Fconnect%252Foauth%252Ftoken%253Fnext%253D%25252F%26scope%3Dglobal%26blog_id%3D0) 
13. **Woo**: [Signup](https://wordpress.com/start/wpcc/oauth2-user?ref=oauth2&oauth2_redirect=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%2F%3Fresponse_type%3Dcode%26client_id%3D50916%26state%3Dc92cfd0e05c2bf6e176a5ed1fbf185a99cf22a08cddf2bd631e1362b240d24f1%26redirect_uri%3Dhttps%253A%252F%252Fwoo.com%252Fwc-api%252Fwpcom-signin%253Fnext%253D%25252Fstart%25252F%252523%25252Finstallation%2526source%253Dsignup_menu%26blog_id%3D0%26wpcom_connect%3D1%26wccom-from%26calypso_env%3Dproduction%26jetpack-code%26jetpack-user-id%3D0%26action%3Doauth2-login%26from-calypso%3D1&oauth2_client_id=50916) and [Login](https://wordpress.com/log-in?redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%2F%3Fresponse_type%3Dcode%26client_id%3D50916%26state%3Dc92cfd0e05c2bf6e176a5ed1fbf185a99cf22a08cddf2bd631e1362b240d24f1%26redirect_uri%3Dhttps%253A%252F%252Fwoo.com%252Fwc-api%252Fwpcom-signin%253Fnext%253D%25252Fstart%25252F%252523%25252Finstallation%2526source%253Dsignup_menu%26blog_id%3D0%26wpcom_connect%3D1%26wccom-from%26calypso_env%3Dproduction%26jetpack-code%26jetpack-user-id%3D0%26action%3Doauth2-login%26from-calypso%3D1&client_id=50916&signup_url=%2Fstart%2Fwpcc%2Foauth2-user%3Fref%3Doauth2%26oauth2_redirect%3Dhttps%253A%252F%252Fpublic-api.wordpress.com%252Foauth2%252Fauthorize%252F%253Fresponse_type%253Dcode%2526client_id%253D50916%2526state%253Dc92cfd0e05c2bf6e176a5ed1fbf185a99cf22a08cddf2bd631e1362b240d24f1%2526redirect_uri%253Dhttps%25253A%25252F%25252Fwoo.com%25252Fwc-api%25252Fwpcom-signin%25253Fnext%25253D%2525252Fstart%2525252F%25252523%2525252Finstallation%252526source%25253Dsignup_menu%2526blog_id%253D0%2526wpcom_connect%253D1%2526wccom-from%2526calypso_env%253Dproduction%2526jetpack-code%2526jetpack-user-id%253D0%2526action%253Doauth2-login%2526from-calypso%253D1%26oauth2_client_id%3D50916)
14. MailPoet: [link](https://wordpress.com/log-in?client_id=79002&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fresponse_type%3Dcode%26redirect_uri%3Dhttps%253A%252F%252Faccount.mailpoet.com%252Flogin%252Fwpcom%252Fcallback%26scope%3Dauth%26client_id%3D79002)

escapemanuele committed:
Login social first, first step

escapemanuele committed:
Updated css

escapemanuele committed:
All buttons working

escapemanuele committed:
Use correct jetpack logo

escapemanuele committed:
Footer of new login

escapemanuele committed:
Fix style

escapemanuele committed:
Remove tos and fix mobile divider

escapemanuele committed:
Add flags

escapemanuele committed:
Navigation link

escapemanuele committed:
Use variable

escapemanuele committed:
Remove comment from scss

crisbusquets commented:
Hi @escapemanuele !

I tested it, and this is what I saw:

`log-in`
- As you mentioned in Slack, some margins/paddings aren't right
- Today I updated the ToS on the Figma file (both the copy and the position).

`log-in/es`
- The magic link CTA overflows the button
- **log in to Wordpress.com** isn't translated (Accede a Wordpress.com)
- **log in with app** needs translation (Acceder a través de la aplicación móvil)
- The same happens with ToS
![image](https://github.com/Automattic/wp-calypso/assets/25607244/c6fdd27c-f488-48ff-a586-8f263f5a5b1c)

`wordpress.com/start/hosting` and `wordpress.com/start/hosting/de`
- Content is identical, but spacings are not. I think the ones we currently have Live are the tight ones.

`wordpress.com/start` and `wordpress.com/start/de`
- Yep, identical!

`wordpress.com/start/videopress-account/user`
- The margins/paddings on the social part are slightly different

escapemanuele committed:
Fix button overflow
Move ToS

escapemanuele committed:
Change button text max width

escapemanuele committed:
Fix social container

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9876551

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @escapemanuele for including a screenshot in the description! This is really helpful for our translators.

crisbusquets commented:
@escapemanuele Aside from the translations I see what I mentioned in my previous comment solved :)

There's one detail the ToS font size is too small (12px). In Figma, it's 14px with a 20px line-height.

Thank you!

escapemanuele committed:
ToS size and height

escapemanuele committed:
Fix width and move css to form

escapemanuele committed:
Fix signup padding

escapemanuele committed:
Fix lost password margin top

escapemanuele committed:
Fix icons

escapemanuele committed:
Fix svg icon

escapemanuele committed:
Fix secondary login pages

escapemanuele committed:
Fix videopress separator

escapemanuele committed:
Fix query args type

escapemanuele committed:
Update ToS content

escapemanuele committed:
Correctly send oauth2client

escapemanuele committed:
Fix id type

escapemanuele committed:
Experiment

escapemanuele committed:
Woo commerce separator fix

escapemanuele committed:
Fix signup padding on mobile

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/9876551).

escapemanuele committed:
Fix woo separator padding

renancarvalho commented on client/login/wp-login/style.scss:
```diff
@@ -448,14 +456,79 @@ $image-height: 47px;
+		margin-top: 24px;
+
+		@include break-mobile {
+			margin-top: 25vh;
```
To better match the proposed design, I think 20.5vh can be used.

alshakero reviewed (changes requested):
### Testing results

#### 2FA screen slightly broken. Extraneous ToS and misaligned divider

![image](https://github.com/Automattic/wp-calypso/assets/17054134/f4840b9d-43d3-4c7f-b49b-8dd46d3e9ddb)

#### Clicking Back on the 2fa screen shows this weird state

![image](https://github.com/Automattic/wp-calypso/assets/17054134/99cbed1b-19ea-4c00-8b0f-db7e4345d210)

#### Clicking use Authenticator app also has the ToS issue

![image](https://github.com/Automattic/wp-calypso/assets/17054134/051f07c3-5337-4f0c-b9ca-9efa7532ace0)


### Email me a login link

✅ Looks great.

### Log in with Passwordless user

✅ Looks great.

### Translation
The heading doesn't seem translated 

![image](https://github.com/Automattic/wp-calypso/assets/17054134/94d20105-005c-4deb-8b65-62c22b5045b2)

escapemanuele committed:
Know if experiment before rendering

renancarvalho commented on client/blocks/login/index.jsx:
```diff
@@ -796,6 +840,8 @@ class Login extends Component {
 
 				{ this.renderNotice() }
 
+				{ this.renderToS() }
```
Shouldn't it be placed after the footer?
here:
<img width="379" alt="image" src="https://github.com/Automattic/wp-calypso/assets/2653810/e6e84ea6-a5b7-4f77-ba2d-192655a290e0">

instead of
<img width="469" alt="image" src="https://github.com/Automattic/wp-calypso/assets/2653810/106bddeb-8142-40d8-9c14-7acabce7d7ed">

escapemanuele replied:
Design is under the header 👍 
4lWBkrFl1jQuN6irMuoLys-fi-306%3A1268

![image](https://github.com/Automattic/wp-calypso/assets/52076348/4fb28ebd-def3-4622-9476-fb5fbf735206)

(You have to look at the third one)

renancarvalho commented on client/login/wp-login/index.jsx:
```diff
@@ -282,7 +290,101 @@ export class Login extends Component {
+			? window.location.origin + pathWithLeadingSlash( this.props.signupUrl )
+			: getSignupUrl( query, currentRoute, oauth2Client, locale, pathname );
+
+		if ( isJetpackCloudOAuth2Client( oauth2Client ) && '/log-in/authenticator' !== currentRoute ) {
```
Maybe we could move this check and the next one up? To avoid this unnecessary check?
```js
const signupUrl = this.props.signupUrl
			? window.location.origin + pathWithLeadingSlash( this.props.signupUrl )
			: getSignupUrl( query, currentRoute, oauth2Client, locale, pathname );
``` 

Line 351 and 355 before 347

escapemanuele replied:
Makes totally sense 👍

heavyweight commented:
<img width="979" alt="Screenshot 2023-12-11 at 14 26 58" src="https://github.com/Automattic/wp-calypso/assets/7000684/c00b364e-6e30-45a4-a52c-a3b584452b5c">
- There is an issue with the left border of the social buttons

- The separator auth-form__separator and form have different sizes on `wordpress.com/start/hosting`

| Before  | After |
| ------------- | ------------- |
| <img width="1535" alt="Screenshot 2023-12-11 at 14 51 04" src="https://github.com/Automattic/wp-calypso/assets/7000684/854e2916-fe94-4ca0-bc0c-badc1ea7e0f8">  | <img width="1523" alt="Screenshot 2023-12-11 at 14 50 54" src="https://github.com/Automattic/wp-calypso/assets/7000684/40a349b7-8499-489e-83e3-c270b8d7ce19"> |


<img width="1920" alt="Screenshot 2023-12-11 at 15 23 36" src="https://github.com/Automattic/wp-calypso/assets/7000684/f0aa568f-62cd-47eb-aa07-dba6f9674b85">
- woo form social buttons are broken

| Before  | After |
| ------------- | ------------- |
|  <img width="1913" alt="Screenshot 2023-12-11 at 15 31 33" src="https://github.com/Automattic/wp-calypso/assets/7000684/53cfa7fd-0dbc-4d3d-9398-671692214f56"> | <img width="1917" alt="Screenshot 2023-12-11 at 15 31 26" src="https://github.com/Automattic/wp-calypso/assets/7000684/1373d62c-fe3a-45a9-aaa2-f250dee4ef4c">  |

We are displaying more links on jetpack login. I just want to confirm that this is the expected behavior.

renancarvalho commented on client/login/wp-login/index.jsx:
```diff
@@ -324,20 +426,43 @@ export class Login extends Component {
+							signupUrl={ signupUrl }
+							usernameOrEmail={ this.state.usernameOrEmail }
+							oauth2Client={ this.props.oauth2Client }
+							getLostPasswordLink={ this.getLostPasswordLink.bind( this ) }
```
Do we really need to `bind` ?

escapemanuele replied:
Sadly, yes, otherwise we would have a problem accessing `this.props` in the LoginLinks arrow component!

renancarvalho commented on client/login/wp-login/index.jsx:
```diff
@@ -371,35 +496,61 @@ export class Login extends Component {
+								<DocumentHead
+									title={ translate( 'Log In' ) }
+									link={ [ { rel: 'canonical', href: canonicalUrl } ] }
+									meta={ [ { name: 'description', content: 'Log in to WordPress.com' } ] }
```
`Log in` or `Log into` 
I have seen both 🤔

escapemanuele replied:
To be honest I preferred not to change this for fear of some SEO problem 😆

heavyweight commented:
P2 social buttons are different but I think this is ok right?

| Before  | After |
| ------------- | ------------- |
|<img width="1915" alt="Screenshot 2023-12-11 at 15 45 49" src="https://github.com/Automattic/wp-calypso/assets/7000684/d402d776-0f72-4b6a-9c09-5b0e37898857">  |  <img width="1920" alt="Screenshot 2023-12-11 at 15 45 56" src="https://github.com/Automattic/wp-calypso/assets/7000684/e0ded37f-6148-433a-ab70-de47cf5e5b55" >|

There is also a slight difference in alignment in `start` which I think is expected as well.
| Before  | After |
| ------------- | ------------- |
| <img width="1920" alt="Screenshot 2023-12-11 at 15 50 11" src="https://github.com/Automattic/wp-calypso/assets/7000684/fdf073a6-6b1b-4e99-a41f-d1e15495ea97"> |  <img width="1920" alt="Screenshot 2023-12-11 at 15 50 17" src="https://github.com/Automattic/wp-calypso/assets/7000684/8a26a1a2-e90b-4579-b5ac-5c11e848adfb"> |

renancarvalho commented on client/login/wp-login/login-buttons.tsx:
```diff
@@ -0,0 +1,134 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
```
Is there a reason to have this component only handling two buttons instead of all the ones available?

Google and Apple I guess are in another place and this handles Jetpack and Magic link?

escapemanuele replied:
It may not be the best way, but in LoginButtons we will have all the additional buttons the we can add after the social ones already there 👍  In this way, the Social component doesn't know about these buttons and can be reused in Signup and in the old Login page as well

renancarvalho reviewed (commented):
Great work 👍 left some comments

escapemanuele committed:
Remove Tos when on 2fa page

escapemanuele committed:
Fix translation missing

escapemanuele committed:
Separator text background

agrullon95 commented:
`/log-in`

* The Google and Apple social login buttons don't have a hover effect vs the application and Email link button. Was that done on purpose?

* The divider line is going through the "OR" text

<img width="331" alt="CleanShot 2023-12-11 at 13 09 49@2x" src="https://github.com/Automattic/wp-calypso/assets/10482592/0da70140-dc0b-4f64-865c-fefb94d6a647">

`/start/hosting`

* The divider line is going through the "OR" text issue from above
* The mobile/tablet viewport (~540px) is missing padding on left & right side
<img width="582" alt="CleanShot 2023-12-11 at 13 15 23@2x" src="https://github.com/Automattic/wp-calypso/assets/10482592/7d774f10-d08c-498e-a365-da9fdf8cd7ac">

`/start/videopress-account/user`

* The divider line is going through the "OR" text issue from above
* Subheader for spanish translation contains extra curly brace in "Log in" link. Reported in i18n channel: p1702316152994339-slack-C02AED43D

alshakero committed:
Merge branch 'trunk' into update/login-social

escapemanuele committed:
Fix videopress background

escapemanuele committed:
Fix new gravatar white login

alshakero commented on client/signup/videopress-step-wrapper/style.scss:
```diff
@@ -237,6 +237,7 @@ body.is-section-signup .videopress-step-wrapper {
 	}
 
 	.auth-form__separator-text {
+		background: none;
```
That's a bit suspicious, the separator test doesn't have a BG, what is setting it so we have to revert it?

escapemanuele committed:
Change margin, move if and address button heigth

alshakero commented on client/blocks/authentication/form-divider/style.scss:
```diff
@@ -29,54 +26,68 @@ $breakpoint-mobile: 660px;
+
+
+.auth-form__separator {
+	background: var(--color-surface);
```
This is it. This will cancel my changes here https://github.com/Automattic/wp-calypso/pull/84551. They were so hard to get right.

escapemanuele replied:
I'll remove these changes!

escapemanuele replied:
Removed ✌️

escapemanuele committed:
Fix double border on woo

escapemanuele commented:
Thank you @heavyweight! 

> The separator auth-form__separator and form have different sizes on wordpress.com/start/hosting

Yes! We have tried to use a standard width for the separator across pages!

> woo form social buttons are broken
> We are displaying more links on jetpack login.
> P2 social buttons are different but I think this is ok right?
> There is also a slight difference in alignment in start which I think is expected as well.

Not anymore! ✌️

escapemanuele committed:
Remove separator background

escapemanuele committed:
Hover effect on social buttons

escapemanuele commented:
Thank you @renancarvalho and @agrullon95, I answered all your questions, I think, and addressed the feedback! ✌️

escapemanuele committed:
Add padding to tablet on signup user

escapemanuele commented:
@alshakero I think I addressed most of your feedback, apart from the screenshot here below.

> Clicking Back on the 2fa screen shows this weird state

I checked, and it is an existing issue. (screenshot is in prod)
I'll try to address this if I can, before merging, or do it in a separate issue.

![image](https://github.com/Automattic/wp-calypso/assets/52076348/6c2ddf24-3e50-4237-987c-409b2f16a826)

escapemanuele committed:
Merge branch 'trunk' into update/login-social

renancarvalho committed:
Fix focus outline style

renancarvalho committed:
Add back outline

escapemanuele committed:
No boxshadow blue on signup button

escapemanuele committed:
Fix focus issue

alshakero reviewed (approved):
Approving just to dismiss my rejection.

heavyweight commented:
The fonts on woo login seem different on desktop and mobile 
Is this expected?

heavyweight committed:
Merge branch 'trunk' into update/login-social

heavyweight committed:
Merge branch 'trunk' into update/login-social

escapemanuele committed:
Exclude desktop app from social first login
